### ISSUE TITLE
Feed with only one item never emits 'readable' or 'data'

### ISSUE THREAD

**Issue by danmactough** [danmactough/node-feedparser#126]
Reported by @comartmit on IRC

Needs more info

**Comment by acreeger** [comment]
The following feed exhibits this issue:
```
<feed xmlns="http://www.w3.org/2005/Atom"><id>http://www.gy.undp.org/content/guyana/en/home/presscenter</id><updated>2015-03-03T05:58:01.536Z</updated><title type="html">Press Center</title><link href="http://www.gy.undp.org/content/guyana/en/home/presscenter.feed" rel="self" /><generator uri="http://www.day.com/cq5" version="5.6">Adobe Experience Manager</generator></feed>
```

**Comment by danmactough** [comment]
Thanks, @acreeger but that feed actually has **no** items.

**Comment by acreeger** [comment]
But "readable" is still never fired. In my mind, it should be fired for every valid feed, but obviously `.read()` should immediately return null. Between listening to `error` and `readable`, you should be able to catch all events, right? Or am I using it differently to how you envisioned?

P.S Thanks for the awesome library.

**Comment by danmactough** [comment]
> But "readable" is still never fired. In my mind, it should be fired for every valid feed, but obviously .read() should immediately return null.

This sounds reasonable. Let me see how it would work in code. @acreeger You have any code in mind?

**Comment by acreeger** [comment]
I was able to workaround this issue by listening to the `end` event, but that only worked because the stream is empty. For a stream with 1 item that exhibits this issue, I believe my workaround would cause that 1 item to be missed.

In terms of code, I don't have any immediate thoughts. Happy to do a bit of work on a PR if we can brainstorm a potential solution. From what I can see, it looks like the readable event is emitted by the underlying `TransformStream`, which inherits that ability from `ReadableStream`. I'm going to assume there's no bug in ReadableStream for now. Perhaps I've got this wrong, but it seems that readable is only emitted by `read`ing the stream - I'm wondering if we're not calling read in some situations? Of course, that hypothesis could be BS given I haven't dug too deep.
